src/plugins/discover/public/application/embeddable/saved_search_embeddable.tsx

```diff
@@ -0,0 +1,410 @@
+
+    props.isLoading = true;
+
+    if (this.savedSearch.grid) {
+      props.settings = this.savedSearch.grid;
+    }
+  }
+
+  private async pushContainerStateParamsToProps(
+    searchProps: SearchProps,
+    { forceFetch = false }: { forceFetch: boolean } = { forceFetch: false }
+  ) {
+    const isFetchRequired =
+      !esFilters.onlyDisabledFiltersChanged(this.input.filters, this.prevFilters) ||
+      !isEqual(this.prevQuery, this.input.query) ||
+      !isEqual(this.prevTimeRange, this.input.timeRange) ||
+      !isEqual(searchProps.sort, this.input.sort || this.savedSearch.sort) ||
+      this.prevSearchSessionId !== this.input.searchSessionId;
+
+    // If there is column or sort data on the panel, that means the original columns or sort settings have
+    // been overridden in a dashboard.
+    searchProps.columns = handleSourceColumnState(
+      { columns: this.input.columns || this.savedSearch.columns },
+      this.services.core.uiSettings
+    ).columns;
+
+    const savedSearchSort =
+      this.savedSearch.sort && this.savedSearch.sort.length
+        ? this.savedSearch.sort
+        : getDefaultSort(
+            this.searchProps?.indexPattern,
+            getServices().uiSettings.get(SORT_DEFAULT_ORDER_SETTING, 'desc')
+          );
+    searchProps.sort = this.input.sort || savedSearchSort;
+    searchProps.sharedItemTitle = this.panelTitle;
+    if (forceFetch || isFetchRequired) {
+      this.filtersSearchSource!.setField('filter', this.input.filters);
+      this.filtersSearchSource!.setField('query', this.input.query);
+      if (this.input.query?.query || this.input.filters?.length) {
+        this.filtersSearchSource!.setField('highlightAll', true);
+      } else {
+        this.filtersSearchSource!.removeField('highlightAll');
+      }
+
+      this.prevFilters = this.input.filters;
+      this.prevQuery = this.input.query;
+      this.prevTimeRange = this.input.timeRange;
+      this.prevSearchSessionId = this.input.searchSessionId;
+      this.searchProps = searchProps;
+      await this.fetch();
+      if (this.node) {
+        await this.rerenderComponent(this.node);
+      }
+    } else if (this.searchProps && this.node) {
+      this.searchProps = searchProps;
+      ReactDOM.render(<SavedSearchEmbeddableComponentMemoized {...this.searchProps} />, this.node);
+    }
+  }
+
+  /**
+   *
+   * @param {Element} domNode
+   */
+  public async render(domNode: HTMLElement) {
+    if (!this.searchProps) {
+      throw new Error('Search props not defined');
+    }
+    if (this.node) {
+      ReactDOM.unmountComponentAtNode(this.node);
+    }
+    this.node = domNode;
+  }
+
+  private async rerenderComponent(domNode: HTMLElement) {
+    if (!this.searchProps) {
+      return;
+    }
+    this.searchProps.refs = domNode;
+    await this.pushContainerStateParamsToProps(this.searchProps);
+    ReactDOM.render(<SavedSearchEmbeddableComponentMemoized {...this.searchProps} />, domNode);
+  }
```

Comment:
I rewrote rendering while debugging the DOM node count, currently in this file there are 2 `ReactDom.render` usages. I rewrote it to his 

```TypeScript
private async renderReactComponent(domNode: HTMLElement, searchProps: SearchProps) {
    return new Promise((res) => {
      ReactDOM.render(<SavedSearchEmbeddableComponentMemoized {...searchProps} />, domNode, res);
    });
  }
```
Was easier to debug, and furthermore, AFAIK react current doesn't render async, but if it would, wrapping it with a Promise would prepare it for this case, also it's cleaner this way IMO